### PR TITLE
fix: improve api migration logic, update migration prompt

### PR DESCRIPTION
**Pull request by jhockett**
#### Description of changes
- Update extensibility migration message to include a note that custom modifications to a CloudFormation template will not be preserved.
- Make REST API migration logic more robust to variable input from `api-params.json` and provide a better error message when no paths are defined.

#### Issue #, if available
https://github.com/aws-amplify/amplify-cli/issues/9225

#### Description of how you validated changes
`apigw-ext-migration.test.ts`  passes

#### Checklist

- [x] PR description included
- [x] `yarn test` passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-category-api/src/index.ts [hunk 1]
```diff
@@ -73,8 +73,9 @@ export async function migrate(context: $TSContext, serviceName?: string) {
       }
     }
   }
-
-  await Promise.all(migrateResourcePromises);
```
Comment (sachscode) [inline packages/amplify-category-api/src/index.ts:77]: Whats the reason for this change ? is it readability ?

**Review by sachscode — COMMENTED** [review]
LGTM , just one question

**Review by sachscode — APPROVED** [review]
LGTM

**Comment by danielleadams** [comment]
I thought I had submitted my review of this, but I think I forgot to click "Submit Review", but I did look at this PR and approve.

**Pull request merged**